clang-tools-extra/clangd/Config.h

```diff
@@ -59,13 +59,15 @@ struct Config {
     std::optional<std::string> FixedCDBPath;
   };
 
+  enum class BuiltinHeaderPolicy { Clangd, QueryDriver };
   /// Controls how the compile command for the current file is determined.
   struct {
     /// Edits to apply to the compile command, in sequence.
     std::vector<llvm::unique_function<void(std::vector<std::string> &) const>>
         Edits;
     /// Where to search for compilation databases for this file's flags.
     CDBSearchSpec CDBSearch = {CDBSearchSpec::Ancestors, std::nullopt};
+    BuiltinHeaderPolicy BuiltinHeaders = BuiltinHeaderPolicy::Clangd;
```

Comment:
Just for consistency, please add a short comment here too (I know we have a more detailed one in `ConfigFragment.h`), e.g.:

```
/// Whether to use clangd's own builtin headers, or ones from the system include extractor
/// if available
```